src/lib.rs

```diff
@@ -209,11 +209,50 @@ pub fn verify_signature<S: Into<String>>(_signature: S, _data: S, _pub_key: S) -
     Err(ErrorCode::FunctionNotImplemented)
 }
 
+#[derive(Serialize, Default)]
+struct CommitInputStruct {
+  entry_type_name: String,
+  entry_content: String,
+}
+
+#[derive(Deserialize, Serialize, Default)]
+struct CommitOutputStruct {
+  hash: String,
+}
 
 /// FIXME DOC
-pub fn commit_entry<S: Into<String>>(_entry_type: S, _entry: serde_json::Value) -> Result<HashString, ErrorCode> {
-    // FIXME
-    Err(ErrorCode::FunctionNotImplemented)
+pub fn commit_entry(_entry_type_name: &str, _entry_content: &str) -> Result<HashString, ErrorCode> {
+    let mut mem_stack: SinglePageStack;
+    unsafe { mem_stack = g_mem_stack.unwrap(); }
+
+    // Put args in struct and serialize into memory
+    let input = CommitInputStruct {
+        entry_type_name: _entry_type_name.to_string(),
+        entry_content: _entry_content.to_string(),
+    };
+    let allocation_of_input =  serialize(&mut mem_stack, input);
+
+    // Call WASMI-able commit
+    let encoded_allocation_of_result: u32;
+    unsafe {
+        encoded_allocation_of_result = hc_commit_entry(allocation_of_input.encode() as u32);
+    }
+    // Check for ERROR in encoding
+    let result = try_deserialize_allocation(encoded_allocation_of_result as u32);
+
+    // what kind of error should we return?
+    // if let Err(e) = result {
```

Comment:
This should work since ErrorCode should be in sync with HcApiReturnCode. Might need to do a cast or a simple conversion function.

Reply:
Yes, they are not currently, we need to add a ticket for syncing them. I also would like to rename most of the errors as they don't read as errors. Like `ErrorActionResult` doesn't tell you what the error is, just where it happened...
I'm fine with returning the error even if they are not synced, because its not this PR's scope.